server/sonar-server/src/main/java/org/sonar/server/qualityprofile/DefinedQProfileRepositoryImpl.java

```diff
@@ -194,7 +223,41 @@ private static boolean ensureAtMostOneDeclaredDefault(Map.Entry<String, List<Def
     }
     MessageDigest md5Digest = DigestUtils.getMd5Digest();
     return builders.stream()
+      .sorted(new SortByParentName(builders))
       .map(builder -> builder.build(md5Digest))
       .collect(MoreCollectors.toList(builders.size()));
   }
+
+  @VisibleForTesting
+  static class SortByParentName implements Comparator<DefinedQProfile.Builder> {
+    private final Map<String, DefinedQProfile.Builder> buildersByName;
+    @VisibleForTesting
+    final Map<String, Integer> depthByBuilder;
+
+    @VisibleForTesting
+    SortByParentName(Collection<DefinedQProfile.Builder> builders) {
+      buildersByName = builders.stream()
+        .collect(MoreCollectors.uniqueIndex(DefinedQProfile.Builder::getName, Function.identity(), builders.size()));
+      Map<String, Integer> depthByBuilder = new HashMap<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "depthByBuilder" which hides the field declared at line 234. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck)